### PR TITLE
Adds multi level metadata documentation

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

File: mule-sdk/v/1.1/multi-level-metadata.adoc [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    return key;
+  }
+
+}
+----
+
+<1> Only the services are retrieved, the actions of a service will be retrieved on demand.
+<2> Build a single `MetadataKey` tree with a new complete level of metadata, in this case, the
+actions level.
+
```
Comment (alegmarra) [inline mule-sdk/v/1.1/multi-level-metadata.adoc]: 3) Add the actions of that service as childs

File: mule-sdk/v/1.1/multi-level-metadata.adoc [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+== Restricted Partial Fetching
```
Comment (alegmarra) [inline mule-sdk/v/1.1/multi-level-metadata.adoc]: Using User Input As Partial Level

File: mule-sdk/v/1.1/multi-level-metadata.adoc [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+*Available since version 1.1*
+
+There are parts of the `MetadataKey` that you may not want to hint the user. This may be
```
Comment (alegmarra) [inline mule-sdk/v/1.1/multi-level-metadata.adoc]: rephrase: There may be cases where you can't hint the user a part of your MetadataKey, like when the universe of options is too big (ie, it makes no sense to show all the classes in a classpath in a dropdown), or when the starting point of the ID is a free input (ie, a query).

File: mule-sdk/v/1.1/multi-level-metadata.adoc [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+because the universe of possibilities it too big, or because it is a custom field. Take as example
+a `MetadataKey` that has a part that is a `String` representing a java class. It would be both very time
+consuming to retrieve all the classes and complicated for the user to have so many possibilities on a
+dropdown.
+
+Because of this you can signal that a `MetadataKeyPart` will not be provided by the resolver and must be
+inserted by the user. This is done by setting to false the `providedByKeyResolver` value on the `MetadataKeyPart`
+annotation.
+
+This is an example where the service name is not given by the resolver :
```
Comment (alegmarra) [inline mule-sdk/v/1.1/multi-level-metadata.adoc]: change and use the java module classes example, it's perfect for showing this

File: mule-sdk/v/1.1/multi-level-metadata.adoc [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+  @Override
+  public String getAction() {
+    return action;
+  }
+}
+----
+
+<1> The service field will have to be inserted by the user without hints.
+
+This also means that on this case the `getKeys` method does not need to return all the
```
Comment (alegmarra) [inline mule-sdk/v/1.1/multi-level-metadata.adoc]: does not need => can't

File: mule-sdk/v/1.1/multi-level-metadata.adoc [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+    for(String action : getActions(key.getService())){
+      key.withChild(MetadataKeyBuilder.newKey(action).build());
+    }
+    return key;
+  }
+
+}
+----
+
+<1> return an empty set of `MetadataKey` since the user will provide this information.
```
Comment (alegmarra) [inline mule-sdk/v/1.1/multi-level-metadata.adoc]: add steps 2 and 3 again to reinforce the idea

**Comment by dukesphere** [comment]
Waiting for approval from Marra

**Pull request merged**